### PR TITLE
refactor: remove unused commonjs plugin

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
- removes unused plugin `@rollup/plugin-commonjs`
### Test Plan:

- [x] CI tests / new tests are not applicable

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
+## [12.0.0-alpha.1](https://github.com/chanzuckerberg/edu-design-system/compare/v12.0.0-alpha.0...v12.0.0-alpha.1) (2023-04-04)
+
```
Comment (jinlee93) [inline CHANGELOG.md:6]: Oop alpha publish changelogs snuck in here but shouldn't matter

File: package.json [hunk 2]
```diff
@@ -109,7 +109,6 @@
     "@commitlint/cli": "^17.5.1",
     "@commitlint/config-conventional": "^17.4.4",
     "@geometricpanda/storybook-addon-badges": "^1.1.1",
-    "@rollup/plugin-commonjs": "^24.0.1",
```
Comment (jinlee93) [inline package.json:112]: Unused plugin, tsconfig handles for us

**Comment by jinlee93** [comment]
All changed component files are just converting `import clsx from "clsx"` to `import { clsx } from "clsx"`. The other changes are alpha publishes and removing unused plugin

File: .storybook/components/TokenSpecimen/TokenSpecimen.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import clsx from 'clsx';
+import { clsx } from 'clsx';
```
Comment (ahuth) [inline .storybook/components/TokenSpecimen/TokenSpecimen.tsx:1]: Weird. What error gets thrown with default export + cjs builds?

Whether it's named or not shouldn't matter.

Comment (booc0mtaco) [reply .storybook/components/TokenSpecimen/TokenSpecimen.tsx:1]: Same question; why is `clsx` having this issue?

Comment (jinlee93) [reply .storybook/components/TokenSpecimen/TokenSpecimen.tsx]: @booc0mtaco paired on this and saw that `<Link>` is being used in a cjs build from `remix-utils` which `requires` clsx, which I thought would pull the `main` build of `clsx` [according to its package.json](https://github.com/lukeed/clsx/blob/master/package.json#L8) but instead uses the `module` build. Removing that `module` line in clsx's package json in edu-stack allows it to resolve to the `main` build instead and works fine with the default export, implying this issue could arise from import/require resolution within edu-stack

Comment (ahuth) [reply .storybook/components/TokenSpecimen/TokenSpecimen.tsx]: Ahhhhhh, the link to remix-utils might be a clue.

I wonder if remix-utils/Link is importing the ESM version of EDS somehow, which then tries to load the ESM versino of clsx? 🤔

Comment (jinlee93) [reply .storybook/components/TokenSpecimen/TokenSpecimen.tsx]: remix-utils/Link is using the CJS version, but 
How about building both cjs and esm for remix-utils and using conditional exports for it?: [aka](https://github.com/chanzuckerberg/frontend-libs/pull/51)
This works on edu-stack

File: CHANGELOG.md (hunk 1 above)
Comment (booc0mtaco) [reply CHANGELOG.md:6]: I looked back and saw a few other older builds included alpha stuff in the change log. This shouldn't interfere with any releases, but makes the change log slightly weird

**Review by booc0mtaco — APPROVED** [review]
simpler PR :) I am assuming this plugin came thru previous work on the feature branch and we are essentially reverting it from there?

**Comment by jinlee93** [comment]
> simpler PR :) I am assuming this plugin came thru previous work on the feature branch and we are essentially reverting it from there?

Yup!

**Comment by jinlee93** [comment]
Going to land this for now since now unrelated to esm / cjs build

**Pull request merged**